lib/response.js

```diff
@@ -638,19 +653,28 @@ exports = module.exports = internals.Response = class {
 
     _close() {
 
-        if (this._processors.close) {
-            try {
-                this._processors.close(this);
+        if (this._state === 'closed') {
+            return;
+        }
+
+        try {
+            if (this._processors.close) {
+                try {
```

Comment:
in general i'm not a fan of nesting `try` blocks, and i'm not totally clear why both are needed here.

what was the condition that was causing the outer block to throw?

Reply:
This is for the `finally` clause which will update the state. It can definitely be done in other ways.

Reply:
If you configure the github diff view to "hide whitespace changes" using the gear icon at the top, it will be more visible what I did.

Reply:
i feel like collapsing these into one `try/catch/finally` is maybe more readable.

though in this case it would have the consequence that if `internals.Response.drain(stream)` were to throw for some reason we would log it with the `response, cleanup, error` tags. the other blocks would have similar concerns that we may need to think about on a case by case basis too.

i don't really think that's a problem though, personally, and it would maintain the existing goal of making sure that the state is updated correctly. what do you think?

Reply:
I think I would rather just set the new state immediately after the initial check. This also means that if the logic somehow recurses to calling itself, it will fail the assert.

Reply:
given that the intent is to set the state regardless of the outcome of the logic in the block, i think that's completely reasonable

Reply:
I have pushed a cleanup patch, which undoes the try-finally and changed the verbiage to use present tense. Hope that resolves this.